python/tk_desktop/install_apps.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import pprint
+
+import sgtk
+
+from sgtk.platform.qt import QtCore
+from sgtk.platform.qt import QtGui
+
+from .ui import install_apps
+
+shotgun_data = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_data")
+
+class InstallApps(QtGui.QWidget):
+    """
+    Widget displayed in the desktop Window's project_commands wiget when no
+    "button" commands are found for the current project. Displays site
+    Software entity icons and a link to a Shotgun Support article detailing
+    how to configure Software entities.
+    """
+    def __init__(self, parent=None):
+        QtGui.QWidget.__init__(self, parent)
+
+        self.ui = install_apps.Ui_InstallApps()
+        self.ui.setupUi(self)
+
+        # Assign magic numbers and such
+        self._icon_size = QtCore.QSize(64, 64)
+        self._icon_layout_spacing = 5
+        self._icon_layout_stretch = 1
+        self._icon_column_count = 4
+        self._config_link_color = "#0000FF"
+        self._config_link_text = "here"
+
+        # Hide this widget by default.
+        self.setVisible(False)
+
+    def build(self, project=None):
```

Comment:
Regarding `project=None` - our current use case always assumes that we are displaying this screen for a particular project. is it OTT to support the `project=None` case? I think we should require a specific project to keep things simple for the time being. We have had issues with this sort of thing in other places, where the logic has gotten a bit out of hand. So I think we should skip the default value for the parameter and on the contrary always require a project id.